### PR TITLE
[Zip][Portable] New package: Gyan.FFmpeg version 5.1.2

### LINKED ISSUE #859: Package Request: ffmpeg
# Description of the new feature/enhancement

ffmpeg is a very useful tool, it would be nice to be able to install it via winget.

# Proposed technical implementation details (optional)

ffmpeg does not however release official builds. The builds available [here](https://ffmpeg.zeranoe.com/builds/) seem to be endorsed on the ffmpeg website, but maybe that's a security issue?

### PR DESCRIPTION
**Pull request by SpecterShell**
* Resolves #859

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/87327)

### CODE REVIEW

**Comment by Trenly** [comment]
@Msftbot .zip

**Review by ItzLevvie — DISMISSED** [review]

**Comment by ItzLevvie** [comment]
@msftbot .zip

**Comment by KaranKad** [comment]
As this build is officially promoted by ffmpeg site should we change `PackageIdentifier` to `FFmpeg.FFmpeg`?

**Comment by SpecterShell** [comment]
> As this build is officially promoted by ffmpeg site should we change `PackageIdentifier` to `FFmpeg.FFmpeg`?

FFmpeg promotes two different builds:

https://www.gyan.dev/ffmpeg/builds/
https://github.com/BtbN/FFmpeg-Builds/releases

So I use this complicated `PackageIdentifier` in case another one would be added to winget-pkgs repo as well.
Don't worry. I have added a moniker `ffmpeg` so that if user types `winget install ffmpeg` it will install this one by default.

**Review by ItzLevvie — DISMISSED** [review]

**Comment by ItzLevvie** [comment]
@msftbot .zip

**Comment by junaga** [comment]
For real, why isn't this merging? This is pipeline is too complicated. Can we _please_ get the package up.

**Comment by Trenly** [comment]
> For real, why isn't this merging? This pipeline is to complicated. Can we _please_ get the package up.

https://github.com/microsoft/winget-pkgs/issues/87300

**Comment by sitiom** [comment]
Please link this PR to #859

File: manifests/g/Gyan/CodexFFmpeg/5.1.2/Gyan.CodexFFmpeg.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+Installers:
+- Architecture: x64
+  NestedInstallerType: portable
+  NestedInstallerFiles:
+  - RelativeFilePath: .\bin\ffmpeg.exe
+    PortableCommandAlias: ffmpeg
+  - RelativeFilePath: .\bin\ffplay.exe
+    PortableCommandAlias: ffplay
+  - RelativeFilePath: .\bin\ffprobe.exe
+    PortableCommandAlias: ffprobe
```
Comment (sitiom) [inline manifests/g/Gyan/CodexFFmpeg/5.1.2/Gyan.CodexFFmpeg.installer.yaml:17]: Fix `RelativeFilePath`:
```suggestion
  - RelativeFilePath: .\ffmpeg-5.1.2-full_build\bin\ffmpeg.exe
    PortableCommandAlias: ffmpeg
  - RelativeFilePath: .\ffmpeg-5.1.2-full_build\bin\ffplay.exe
    PortableCommandAlias: ffplay
  - RelativeFilePath: .\ffmpeg-5.1.2-full_build\bin\ffprobe.exe
    PortableCommandAlias: ffprobe
```

**Review by sitiom — CHANGES_REQUESTED** [review]

**Review by sitiom — CHANGES_REQUESTED** [review]
https://github.com/GyanD/codexffmpeg/issues/64#issuecomment-1406424272:
> @GyanD
> The pkg name should be just ffmpeg, not codexffmpeg. The latter is the name of the site.

**Review by Trenly — APPROVED** [review]

**Review by stephengillie — APPROVED** [review]

**Comment by rlaphoenix** [comment]
After installation the aliases do not work for me. Tried many restarts, pwsh, powershell, and cmd. None work.
The environment path added was `C:\Users\...\AppData\Local\Microsoft\WinGet\Packages\Gyan.FFmpeg_Microsoft.Winget.Source_8wekyb3d8bbwe` but it needs to be `C:\Users\Justin\AppData\Local\Microsoft\WinGet\Packages\Gyan.FFmpeg_Microsoft.Winget.Source_8wekyb3d8bbwe\ffmpeg-5.1.2-full_build\bin` before it works.

**Comment by rlaphoenix** [comment]
After installation the aliases do not work for me. Tried many restarts, pwsh, powershell, and cmd. None work.
The environment path added was `C:\Users\...\AppData\Local\Microsoft\WinGet\Packages\Gyan.FFmpeg_Microsoft.Winget.Source_8wekyb3d8bbwe` but it needs to be `C:\Users\...\AppData\Local\Microsoft\WinGet\Packages\Gyan.FFmpeg_Microsoft.Winget.Source_8wekyb3d8bbwe\ffmpeg-5.1.2-full_build\bin` before it works.

**Comment by mstevens83** [comment]
After installation this does not work for me. It seems like the aliases are incorrect.
@rlaphoenix reported the same issue but it seems his post got deleted?

**Comment by mstevens83** [comment]
Hi rlaphoenix, I have the same issue. Your post is deleted though, did you
fix the issue yourself? I'm interested to know how.
Regards, Matthias

Op do 2 feb. 2023 om 00:59 schreef rlaphoenix ***@***.***>:

**Comment by sitiom** [comment]
Works fine.
image

**Comment by rlaphoenix** [comment]
> After installation this does not work for me. It seems like the aliases are incorrect. @rlaphoenix reported the same issue but it seems his post got deleted?

I deleted as I moved discussion to https://github.com/microsoft/winget-pkgs/issues/95349 and I invite you guys to as well.

**Pull request merged**